### ISSUE TITLE
Admin: Tools - API Keys - Enter username field needs to search all WP users

### ISSUE THREAD

**Issue by gspice** [AffiliateWP/AffiliateWP#2749]
Currently in the WordPress admin when you navigate to _Affiliates - Tools - API Keys_  and begin to type in the `Enter username` field, it will only search affiliate accounts.  The change requested is it needs to search **all** WP user accounts.   As it stands, you do _**not**_ have to be an affiliate to generate a set of API keys, but this leads to errors down the line.  For example, if you do not enter the username correctly, this results in a 401 connection error when attempting to connect with Zapier using the API credentials.

**Comment by DrewAPicture** [comment]
Yeah, I'd consider this a bug. If I remember correctly, I recall there being some question of who potential API consumers would be, and ultimately we left it as is. Realistically speaking, _any_ user could conceivably be a valid REST consumer, there's just some capabilities that would have to be conveyed to make big parts of the API usable to them – worth noting that affiliates don't automatically get those capabilities either, thus a bug not an enhancement.
